### PR TITLE
feat(notifications): Move notifications socket to Android native

### PR DESCRIPTION
**Pull request by yoav97**
## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: android/app/src/main/java/chat/rocket/android/MainApplication.java [hunk 1]
```diff
@@ -29,6 +30,12 @@
 
 public class MainApplication extends Application implements ReactApplication {
 
+  private static Context context;
```
Comment (harelush) [inline android/app/src/main/java/chat/rocket/android/MainApplication.java:33]: saving context as static is a bad practice and can cause data leaks

File: android/app/src/main/java/chat/rocket/android/messages/NotificationPresenter.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            NotificationChannel channel = new NotificationChannel(CHANNEL_ID, "NOTIFICATION", NotificationManager.IMPORTANCE_HIGH);
+            channel.setDescription("NOTIFICATION NOTIFICATION CHANNEL");
+            channel.enableLights(true);
+            channel.enableVibration(true);
+            notificationManager.createNotificationChannel(channel);
+        }
+    }
+
+    // TODO: Check if data passing works
```
Comment (harelush) [inline android/app/src/main/java/chat/rocket/android/messages/NotificationPresenter.java:50]: how should we check it?

File: android/app/src/main/java/chat/rocket/android/messages/NotificationClient.java [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+            Bundle data = new Bundle();
+            data.putString("rid", rid);
+            data.putString("name", name);
+            data.putString("sender", senderUsername);
+            data.putString("messageId", messageId);
+            data.putString("messageType", messageType);
+            data.putString("type", type);
+            data.putString("host", host);
+
+            // TODO: Fix is background
```
Comment (harelush) [inline android/app/src/main/java/chat/rocket/android/messages/NotificationClient.java:191]: fixed?

**Comment by harelush** [comment]
@yoav97 please check commens

File: android/app/src/main/java/chat/rocket/android/MainApplication.java (hunk 1 above)
Comment (yoav97) [reply android/app/src/main/java/chat/rocket/android/MainApplication.java:33]: Discussed F2F

File: android/app/src/main/java/chat/rocket/android/messages/NotificationPresenter.java (hunk 2 above)
Comment (yoav97) [reply android/app/src/main/java/chat/rocket/android/messages/NotificationPresenter.java:50]: Discussed F2F

File: android/app/src/main/java/chat/rocket/android/messages/NotificationClient.java (hunk 3 above)
Comment (yoav97) [reply android/app/src/main/java/chat/rocket/android/messages/NotificationClient.java:191]: Removed

File: android/app/src/main/java/chat/rocket/android/MainActivity.java [hunk 4]
```diff
@@ -252,4 +253,11 @@ else if (requestCode == REQUEST_BACKGROUND_PERMISSIONS_REQUEST_CODE)
                 permissions,
                 requestCode);
     }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
+
+        NotificationsService.startService(this);
```
[the remark this answers is not in the document]
Comment (yoav97) [reply android/app/src/main/java/chat/rocket/android/MainActivity.java:261]: Discussed F2F

**Pull request merged**